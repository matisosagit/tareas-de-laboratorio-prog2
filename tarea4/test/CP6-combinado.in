crearCP 10

# la cola está vacía
estaEnCP 0
estaEnCP 1
esVaciaCP

# creo Perro 1 y agrego a cola
crearFecha 2/10/2022
crearPerro 1 Rocky 3 1 Un Beagle amigable
insertarEnCP

# ahora la cola no está vacía
esVaciaCP

# el prioritario es el Perro 1
prioritarioCP

# la vitalidad de Perro 1 es 1
prioridadEnCP 1

# elimino el perro prioritario
eliminarPrioritarioCP
esVaciaCP # Verifico que la cola queda vacía

# creo Perro 1 (nuevo, ID 1 ya que la cola estaba vacía) y agrego a cola
crearFecha 10/10/2022
crearPerro 1 Luna 3 1 Una Labradora juguetona
insertarEnCP

# ahora el prioritario es el Perro 1 (el nuevo)
prioritarioCP

# ahora elimino el Perro 1
eliminarPrioritarioCP
esVaciaCP # Verifico que la cola queda vacía

# agrego 10 perros a la cola. Las vitalidades se asignan para que
# Perro 10 (fecha 10/10/2021) sea el prioritario (Vitalidad 1),
# luego Perro 1 (fecha 1/10/2022, Vitalidad 2),
# luego Perro 9 (fecha 3/10/2022, Vitalidad 3), y así sucesivamente.

# creo Perro 1 (Vitalidad 2)
crearFecha 1/10/2022
crearPerro 1 Max 3 2 Un Caniche inteligente
insertarEnCP

# creo Perro 2 (Vitalidad 10)
crearFecha 10/10/2022
crearPerro 2 Coco 3 10 Un Bulldog cariñoso
insertarEnCP

# creo Perro 3 (Vitalidad 9)
crearFecha 9/10/2022
crearPerro 3 Toby 3 9 Un Pastor Alemán leal
insertarEnCP

# creo Perro 4 (Vitalidad 8)
crearFecha 8/10/2022
crearPerro 4 Bruno 3 8 Una Husky energética
insertarEnCP

# creo Perro 5 (Vitalidad 7)
crearFecha 7/10/2022
crearPerro 5 Lola 3 7 Un Boxer valiente
insertarEnCP

# creo Perro 6 (Vitalidad 6)
crearFecha 6/10/2022
crearPerro 6 Kira 3 6 Un perro adorable
insertarEnCP

# creo Perro 7 (Vitalidad 5)
crearFecha 5/10/2022
crearPerro 7 Simba 3 5 Una Golden Retriever dócil
insertarEnCP

# creo Perro 8 (Vitalidad 4)
crearFecha 4/10/2022
crearPerro 8 Nala 3 4 Un Chihuahua pequeño
insertarEnCP

# creo Perro 9 (Vitalidad 3)
crearFecha 3/10/2022
crearPerro 9 Zeus 3 3 Una Poodle elegante
insertarEnCP

# creo Perro 10 (Vitalidad 1 - el prioritario de este grupo)
# Su prioridad se basa en la fecha 10/10/2021
crearFecha 10/10/2021
crearPerro 10 Otto 3 1 Un Doberman protector
insertarEnCP

# el prioritario es el Perro 10 (Vitalidad 1)
prioritarioCP

# la vitalidad del Perro 1 es 2
prioridadEnCP 1

# la vitalidad del Perro 10 es 1
prioridadEnCP 10

# verifico que algunos perros estén en la cola
estaEnCP 1
estaEnCP 2
estaEnCP 10

# verifico las vitalidades nuevamente
prioridadEnCP 1
prioridadEnCP 2
prioridadEnCP 10

# elimino el prioritario (Perro 10, Vit 1)
eliminarPrioritarioCP
# ahora el prioritario es el Perro 1 (Vitalidad 2)
prioritarioCP

# elimino el prioritario (Perro 1, Vit 2)
eliminarPrioritarioCP
# ahora el prioritario es el Perro 9 (Vitalidad 3)
prioritarioCP

liberarCP

Fin