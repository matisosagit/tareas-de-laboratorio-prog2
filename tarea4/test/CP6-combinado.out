1> Se creó una cola de prioridad de perros con cantidad maxima 10
2> #  la cola está vacía.
3> El perro con id 0 NO está en la cola de prioridad.
4> El perro con id 1 NO está en la cola de prioridad.
5> La cola de prioridad es vacía.
6> #  creo Perro 1 y agrego a cola.
7> Fecha creada en forma exitosa.
8> 9> Se insertó el perro en la cola de prioridad.
10> #  ahora la cola no está vacía.
11> La cola de prioridad NO es vacía.
12> #  el prioritario es el Perro 1.
13> Perro 1
Nombre: Rocky
Edad: 3
Descripcion:  Un Beagle amigable
Fecha de ingreso: 2/10/2022
Vitalidad: 1
14> #  la vitalidad de Perro 1 es 1.
15> La prioridad del perro de id 1 es 1
16> #  elimino el perro prioritario.
17> Se eliminó el elemento prioritario de la cola de prioridad.
18> La cola de prioridad es vacía.
19> #  creo Perro 1 (nuevo, ID 1 ya que la cola estaba vacía) y agrego a cola.
20> Fecha creada en forma exitosa.
21> 22> Se insertó el perro en la cola de prioridad.
23> #  ahora el prioritario es el Perro 1 (el nuevo).
24> Perro 1
Nombre: Luna
Edad: 3
Descripcion:  Una Labradora juguetona
Fecha de ingreso: 10/10/2022
Vitalidad: 1
25> #  ahora elimino el Perro 1.
26> Se eliminó el elemento prioritario de la cola de prioridad.
27> La cola de prioridad es vacía.
28> #  agrego 10 perros a la cola. Las vitalidades se asignan para que.
29> #  Perro 10 (fecha 10/10/2021) sea el prioritario (Vitalidad 1),.
30> #  luego Perro 1 (fecha 1/10/2022, Vitalidad 2),.
31> #  luego Perro 9 (fecha 3/10/2022, Vitalidad 3), y así sucesivamente..
32> #  creo Perro 1 (Vitalidad 2).
33> Fecha creada en forma exitosa.
34> 35> Se insertó el perro en la cola de prioridad.
36> #  creo Perro 2 (Vitalidad 10).
37> Fecha creada en forma exitosa.
38> 39> Se insertó el perro en la cola de prioridad.
40> #  creo Perro 3 (Vitalidad 9).
41> Fecha creada en forma exitosa.
42> 43> Se insertó el perro en la cola de prioridad.
44> #  creo Perro 4 (Vitalidad 8).
45> Fecha creada en forma exitosa.
46> 47> Se insertó el perro en la cola de prioridad.
48> #  creo Perro 5 (Vitalidad 7).
49> Fecha creada en forma exitosa.
50> 51> Se insertó el perro en la cola de prioridad.
52> #  creo Perro 6 (Vitalidad 6).
53> Fecha creada en forma exitosa.
54> 55> Se insertó el perro en la cola de prioridad.
56> #  creo Perro 7 (Vitalidad 5).
57> Fecha creada en forma exitosa.
58> 59> Se insertó el perro en la cola de prioridad.
60> #  creo Perro 8 (Vitalidad 4).
61> Fecha creada en forma exitosa.
62> 63> Se insertó el perro en la cola de prioridad.
64> #  creo Perro 9 (Vitalidad 3).
65> Fecha creada en forma exitosa.
66> 67> Se insertó el perro en la cola de prioridad.
68> #  creo Perro 10 (Vitalidad 1 - el prioritario de este grupo).
69> #  Su prioridad se basa en la fecha 10/10/2021.
70> Fecha creada en forma exitosa.
71> 72> Se insertó el perro en la cola de prioridad.
73> #  el prioritario es el Perro 10 (Vitalidad 1).
74> Perro 10
Nombre: Otto
Edad: 3
Descripcion:  Un Doberman protector
Fecha de ingreso: 10/10/2021
Vitalidad: 1
75> #  la vitalidad del Perro 1 es 2.
76> La prioridad del perro de id 1 es 2
77> #  la vitalidad del Perro 10 es 1.
78> La prioridad del perro de id 10 es 1
79> #  verifico que algunos perros estén en la cola.
80> El perro con id 1 está en la cola de prioridad.
81> El perro con id 2 está en la cola de prioridad.
82> El perro con id 10 está en la cola de prioridad.
83> #  verifico las vitalidades nuevamente.
84> La prioridad del perro de id 1 es 2
85> La prioridad del perro de id 2 es 10
86> La prioridad del perro de id 10 es 1
87> #  elimino el prioritario (Perro 10, Vit 1).
88> Se eliminó el elemento prioritario de la cola de prioridad.
89> #  ahora el prioritario es el Perro 1 (Vitalidad 2).
90> Perro 1
Nombre: Max
Edad: 3
Descripcion:  Un Caniche inteligente
Fecha de ingreso: 1/10/2022
Vitalidad: 2
91> #  elimino el prioritario (Perro 1, Vit 2).
92> Se eliminó el elemento prioritario de la cola de prioridad.
93> #  ahora el prioritario es el Perro 9 (Vitalidad 3).
94> Perro 9
Nombre: Zeus
Edad: 3
Descripcion:  Una Poodle elegante
Fecha de ingreso: 3/10/2022
Vitalidad: 3
95> Se liberó la cola de prioridad de perros
96> Fin.
